### ISSUE TITLE
Union function failes in bulk_udpate_or_create

### ISSUE THREAD

**Issue by M1hacka** [M1hacka/django-pg-bulk-update#23]
```
ProgrammingError('operator does not exist: integer[] || bigint[]\nLINE 4: ...UNNEST(COALESCE("apps_appstats"."user_ids", \'{}\') || "vals"....\n                                                             ^\nHINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.\n',)
Traceback (most recent call last):
  File "/home/vagrant/venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
psycopg2.ProgrammingError: operator does not exist: integer[] || bigint[]
LINE 4: ...UNNEST(COALESCE("apps_appstats"."user_ids", '{}') || "vals"....
                                                             ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/vagrant/venv/lib/python3.6/site-packages/celery/app/trace.py", line 382, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/vagrant/venv/lib/python3.6/site-packages/celery/app/trace.py", line 641, in __protected_call__
    return self.run(*args, **kwargs)
  File "/vagrant/cqapi/cqapi/celery.py", line 192, in run_wrapper
    return self.task_body(*args, **kwargs)
  File "/vagrant/cqapi/apps/tasks.py", line 151, in process_app_stats
    AppStats.process_redis_stats()
  File "/vagrant/cqapi/apps/models.py", line 997, in process_redis_stats
    set_functions=set_functions)
  File "/home/vagrant/venv/lib/python3.6/site-packages/django_pg_bulk_update/manager.py", line 140, in bulk_update_or_create
    batch_size=batch_size, batch_delay=batch_delay)
  File "/home/vagrant/venv/lib/python3.6/site-packages/django_pg_bulk_update/query.py", line 703, in bulk_update_or_create
    data_arg_index=1, batch_size=batch_size, batch_delay=batch_delay)
  File "/home/vagrant/venv/lib/python3.6/site-packages/django_pg_bulk_update/utils.py", line 175, in batched_operation
    r = handler(*args, **kwargs)
  File "/home/vagrant/venv/lib/python3.6/site-packages/django_pg_bulk_update/query.py", line 632, in _insert_on_conflict_no_validation
    cursor.execute(val_sql + upd_sql, params=val_params + upd_params)
  File "/home/vagrant/venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 100, in execute
    return super().execute(sql, params)
  File "/home/vagrant/venv/lib/python3.6/site-packages/cacheops/transaction.py", line 99, in execute
    result = self._no_monkey.execute(self, sql, params)
  File "/home/vagrant/venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/vagrant/venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/vagrant/venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "/home/vagrant/venv/lib/python3.6/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/vagrant/venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: operator does not exist: integer[] || bigint[]
LINE 4: ...UNNEST(COALESCE("apps_appstats"."user_ids", '{}') || "vals"....
                                                             ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.

```

**Comment by M1hacka** [comment]
The bug is in ConcatSetFunction which is used by UnionSetFunction

**Comment by M1hacka** [comment]
Can't generate on tests, need more info about executed query from target project.

**Comment by M1hacka** [comment]
Released 2.1.0
